app/perfsymboltable.cpp

```diff
@@ -920,6 +920,15 @@ Dwfl *PerfSymbolTable::attachDwfl(void *arg)
     if (static_cast<pid_t>(m_pid) == dwfl_pid(m_dwfl))
         return m_dwfl; // Already attached, nothing to do
 
+    //  Fix for issue: dwfl_attach_state should be called only for user-level CPU register state
+    //  and user-level stack, allowing stack unwinding.
+    PerfUnwind::UnwindInfo *unwindInfo = static_cast<PerfUnwind::UnwindInfo *>(arg);
+    if (!((unwindInfo->sample->type() & PERF_SAMPLE_REGS_USER) &&
+          // Records the current user-level CPU register state
+          (unwindInfo->sample->type() & PERF_SAMPLE_STACK_USER)))
```

Comment:
this part shouldn't be checked here, it will break analysis of data files recorded without `--call-graph dwarf` - we still want dwfl setup then to analyze inline frames and symbols

Reply:
if this part is needed, then maybe move it to `PerfUnwind::analyze` - but I don't think we should need it - do we really get a `userStack().length() > 0` without `PERF_SAMPLE_STACK_USER`? I doubt that

Reply:
This check did not break  analysis of data files recorded without --call-graph dwarf. 
We simply disable the erroneous call to the dwfl_attach_state function. And this check should be done right here, after the if statement if (static_cast <pid_t> (m_pid) == dwfl_pid (m_dwfl)). In this case, the check does not violate the analysis in the resolveCallchain function, and moreover, unwindStack. Moreover, this check is performed only once for each pid when attachDwfl is called from unwindStack and only unnecessary calls from resolveCallchain are intercepted by this check.

Reply:
I just looked at the checks when calling the dwfl_attach_state function in linux perf.
Perf report does the same check: machine.c, line 2495 -
if (! ((evsel-> core.attr.sample_type & PERF_SAMPLE_REGS_USER) &&
       (evsel-> core.attr.sample_type & PERF_SAMPLE_STACK_USER)))
return 0;